src/Components/CreateImageWizard/formComponents/AzureSourcesSelect.js

```diff
@@ -49,7 +49,13 @@ const AzureSourcesSelect = ({ label, isRequired, className, ...props }) => {
     if (isFetchingDetails || !isSuccessDetails) return;
     change('azure-tenant-id', sourceDetails?.azure?.tenant_id);
     change('azure-subscription-id', sourceDetails?.azure?.subscription_id);
-  }, [isFetchingDetails, isSuccessDetails]);
+  }, [
+    isFetchingDetails,
+    isSuccessDetails,
+    sourceDetails?.azure?.subscription_id,
+    sourceDetails?.azure?.tenant_id,
+    change,
+  ]);
```

Comment:
Use this pattern instead:
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

It's not a good approach to make it dependent to the `change` function and the other you added.

Reply:
as we discussed in the group channel, we agreed to add those dependencies
and in the future maybe we will refector the code and remove the useEffect hook were ever we can